File: Solutions in Java/ReverseLinkedList.java
/**
 * @author : Anmol Agrawal
 */
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        if(head == null){
            return head;
        }
        ListNode curr = head;
        while(curr.next != null){
            ListNode temp = curr.next;
            curr.next = temp.next;
            temp.next = head;
            head = temp;
        }
        return head;
    }
}

File: Solutions in Javascript/ReverseLinkedList.js
/**
 * @author : Anmol Agrawal
 */
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    if(head === null){
        return head;
    }
    let curr = head;
    while(curr.next != null){
        let temp = curr.next;
        curr.next = temp.next;
        temp.next = head;
        head = temp;
    }
    return head;
};